glean-core/src/upload/mod.rs

```diff
@@ -98,22 +98,32 @@ impl PingUploadManager {
     }
 
     /// Creates a `PingRequest` and adds it to the queue.
-    pub fn enqueue_ping(&self, uuid: &str, url: &str, body: JsonValue) {
+    pub fn enqueue_ping(&self, uuid: &str, path: &str, body: JsonValue) {
         let mut queue = self
             .queue
             .write()
             .expect("Can't write to pending pings queue.");
-        let request = PingRequest::new(uuid, url, body);
+        let request = PingRequest::new(uuid, path, body);
         queue.push_back(request);
     }
 
-    /// Clears the pending pings queue.
-    pub fn clear_ping_queue(&self) {
+    /// Clears the pending pings queue, leaves the deletion-request pings.
+    pub fn clear_ping_queue(&self) -> RwLockWriteGuard<'_, VecDeque<PingRequest>> {
```

Comment:
Why do we need to return the lock guard?

Reply:
So that we can acquire it when clearing all the stores. Here: https://github.com/mozilla/glean/pull/729/files#diff-baecdb62c1f9893931c0b7b7b9a4a947R334

Reply:
Oh I see.